src/phase0/state_transition/block_processing.rs

```diff
@@ -160,16 +163,38 @@ fn process_block_header<
         MAX_VALIDATORS_PER_COMMITTEE,
         PENDING_ATTESTATIONS_BOUND,
     >,
-    _block: &mut BeaconBlock<
+    block: &mut BeaconBlock<
         MAX_PROPOSER_SLASHINGS,
         MAX_VALIDATORS_PER_COMMITTEE,
         MAX_ATTESTER_SLASHINGS,
         MAX_ATTESTATIONS,
         MAX_DEPOSITS,
         MAX_VOLUNTARY_EXITS,
     >,
-    _context: &Context,
+    context: &Context,
 ) -> Result<(), Error> {
+    if block.slot != state.slot {
+        return Err(Error::InvalidHeader(InvalidHeader::StateSlotMismatch));
+    }
+
+    let expected_block_root = state.latest_block_header.hash_tree_root()?;
+    if block.parent_root != expected_block_root {
+        return Err(Error::InvalidHeader(InvalidHeader::ParentBlockRootMismatch));
+    }
+
+    state.latest_block_header = crate::phase0::beacon_block::BeaconBlockHeader {
```

Comment:
```suggestion
    state.latest_block_header = BeaconBlockHeader {
```

and up above let's just import it as a single symbol:

`use crate::phase0::beacon_block::BeaconBlockHeader`

Reply:
Done 7b8dfbf